Fix Depth Outlines for URP 13+

CaseyHofland opened the pull request:
Fixes #60

Outlines would trigger a CommandBuffer warning (full error below) when "Enable Depth Testing" was enabled, turning the screen black.

For whatever reason, `new RenderTargetIdentifier("_CameraDepthTexture")` don't work no mo and Unity decided to provide no solid alternative (you could use camData.renderer.cameraDepthTargetHandle but it suffers from incompatibility with MSAA). THE ONLY FIX I could find was hidden in [some desolate forum](https://forum.unity.com/threads/how-to-set-_cameradepthtexture-as-render-target-in-urp-13.1279934/#post-8272821) using Reflection, thank god for the community.

Despite my attitude I'm really happy with this fix: this is the best Outline package out there (...right?) and I would hate for it to get abandoned because of Unity pullin' a fast one on us.

```
CommandBuffer: temporary render texture _CameraDepthTexture not found while executing Outline--1 (SetRenderTarget depth buffer)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)
```

_sidenote: the diff is larger than it should be, probably because I updated line endings. Oh well 🤷_

CaseyHofland committed:
Fix Depth Outlines for URP 13+

Outlines would trigger a CommandBuffer warning (full error below) when "Enable Depth Testing" was enabled, turning the screen black.

For whatever reason, `new RenderTargetIdentifier("_CameraDepthTexture")` don't work no mo and Unity decided to provide no solid alternative (you could use camData.renderer.cameraDepthTargetHandle but it suffers from incompatibility with MSAA). THE ONLY FIX I could find was hidden in [some desolate forum](https://forum.unity.com/threads/how-to-set-_cameradepthtexture-as-render-target-in-urp-13.1279934/#post-8272821) using Reflection, thank god for the community.

Despite my attitude I'm really happy with this fix: this is the best Outline package out there (...right?) and I would hate for it to get abandoned because of Unity pullin' a fast one on us.

```
CommandBuffer: temporary render texture _CameraDepthTexture not found while executing Outline--1 (SetRenderTarget depth buffer)
UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)
```

Arvtesh reviewed (changes requested):
Please add comments with a link to the workaround description on forums and it looks like indentation is messed up.

CaseyHofland commented:
Weird, I don’t have any indentation issues on my system.
I assume you’re not talking about the directives aligned on the far left. My understanding is that that’s standard practice.

CaseyHofland committed:
Update OutlinePass.cs

Added a clarification comment on line 59 explaining why this fix was implemented.
If the explanation is too extensive to be useful it may be changed to something like "For motivation of using reflection, see https://forum.unity.com/threads/how-to-set-_cameradepthtexture-as-render-target-in-urp-13.1279934/#post-8272821"

james-arndt commented:
This is still an issue. URP 14.0.8 using Unity 2022.3.6f1. Same exact error seen above when enabling "Enable Depth Testing".
